src/Microsoft.TemplateEngine.Cli/New3Command.cs

```diff
@@ -180,6 +178,11 @@ private IFilteredTemplateInfo UseSecondaryCriteriaToDisambiguateTemplateMatches(
 
         public static int Run(string commandName, ITemplateEngineHost host, Action<IEngineEnvironmentSettings, IInstaller> onFirstRun, string[] args)
         {
+            if (args.Any(x => string.Equals(x, "--debug:attach", StringComparison.Ordinal)))
+            {
+                Console.ReadLine();
```

Comment:
Might be useful to check `System.Diagnostics.Debugger.IsAttached` instead of `Console.ReadLine()`, incase this command is not run from console.